### DISCUSSION TITLE
Objective function and marginal hourly cost question

### DISCUSSION THREAD

**Discussion by PeterKlein11** [blue-marble/gridpath#1132] [Q&A] [answered]
Hi All

I am working through the examples. Judging by the sign of the objective function I am guessing negative indicates cost?

I want to look at the hourly system marginal costs, which I am assuming are found under system_load_zone_timepoint.csv in the results?

When I look at the hourly marginal system cost I see that when there is a violation for example overgeneration or unserved energy the marginal cost is shown with the opposite +ve sign compared to other -ve signs. Could anyone explain this behaviour?

Regards
Peter

**Comment by anamileva** [comment, marked as answer]
Hi @PeterKlein11.

> I am working through the examples. Judging by the sign of the objective function I am guessing negative indicates cost?

The GridPath objective function is quite flexible, so that it can accommodate cost minimization, profit maximization, a combination of both, etc. Negative values in the example objective functions indicate costs.

> I want to look at the hourly system marginal costs, which I am assuming are found under system_load_zone_timepoint.csv in the results?

That is correct.

> When I look at the hourly marginal system cost I see that when there is a violation for example overgeneration or unserved energy the marginal cost is shown with the opposite +ve sign compared to other -ve signs. Could anyone explain this behaviour?

You should only be seeing the opposite sign when there is overgeneration. This is because when we add load, we will experience less overgeneration and therefore lower costs (positive sign). When unserved energy is being experienced, additional load will lead to higher costs (negative sign) and you should be seeing negative marginal values.
